### DISCUSSION TITLE
Generating a signed certificate beforehand

### DISCUSSION THREAD

**Discussion by jorants** [mitmproxy/mitmproxy#5665] [Intercept Everything!]
Hi,

I am trying to listen in on an Android app that ignored the CA files in the system store and ships its own `.cer` files in the apk. `apk-mitm` was not successful as the app seems to implement it's own logic. Hence my question: how do I generate a public key file/dummy cert for a domain that I can then drop into the APK and that will match the one used by mitmproxy? Does mitm proxy store the generated dummy certificates for different domains somewhere?

**Comment by mhils** [comment]
It does not store the dummy certs, but you can instruct mitmproxy to use a certificate you generated externally: https://docs.mitmproxy.org/stable/concepts-certificates/#using-a-custom-server-certificate
